coffee_maker_navigator_2/lib/app.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:coffee_maker_navigator_2/di/di/dependency_container_manager.dart';
+import 'package:coffee_maker_navigator_2/di/di/dependency_containers/app_level_dependency_container.dart';
+import 'package:coffee_maker_navigator_2/di/di/injector.dart';
+import 'package:coffee_maker_navigator_2/ui/router/view/app_router_delegate.dart';
+import 'package:coffee_maker_navigator_2/ui/router/view_model/router_view_model.dart';
+import 'package:demo_ui_components/demo_ui_components.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class CoffeeMakerApp extends StatefulWidget {
+  const CoffeeMakerApp({Key? key}) : super(key: key);
+
+  @override
+  State<CoffeeMakerApp> createState() => _CoffeeMakerAppState();
+}
+
+class _CoffeeMakerAppState extends State<CoffeeMakerApp> {
+  late final AppRouterDelegate _appRouterDelegate;
+  late final BackButtonDispatcher _backButtonDispatcher;
+  late final DependencyContainerManager _containerManager;
+
+  @override
+  void initState() {
+    super.initState();
+    _appRouterDelegate = AppRouterDelegate();
+    _backButtonDispatcher = RootBackButtonDispatcher();
+    _containerManager = DependencyContainerManager();
```

Comment:
maybe better to use consistently .instance to avoid thoughts that here is a separate instance created

Reply:
done fe4bdaa14d71579319b4d7386c18390e993e36c0